coalib/settings/ConfigurationGathering.py

```diff
@@ -394,13 +394,14 @@ def gather_configuration(acquire_settings,
         arg_list = sys.argv[1:] if arg_list is None else arg_list
     sections, targets = load_configuration(arg_list, log_printer, arg_parser,
                                            args=args)
-    local_bears, global_bears = fill_settings(sections,
-                                              acquire_settings,
-                                              log_printer)
+    local_bears, global_bears, aspects = fill_settings(sections,
+                                                       acquire_settings,
+                                                       log_printer)
     save_sections(sections)
     warn_nonexistent_targets(targets, sections, log_printer)
 
     return (sections,
             local_bears,
             global_bears,
+            aspects,
```

Comment:
a tuple return value for a public API is always bad news for extensibility.

one way to work around that is ...

You can create a new method `fill_settings_new` & `gather_configuration_new` (obviously use a different naming convention) which uses the existing `fill_settings` & `gather_configuration` but also does additional logic and has different return values.